drivers/display/display_ili9xxx.c

```diff
@@ -142,6 +185,93 @@ static int ili9xxx_write(const struct device *dev, const uint16_t x,
 	return 0;
 }
 
+#ifdef CONFIG_ILI9XXX_READ
+
+static int ili9xxx_read(const struct device *dev, const uint16_t x,
+			const uint16_t y,
+			const struct display_buffer_descriptor *desc, void *buf)
+{
+	const struct ili9xxx_config *config = dev->config;
+	struct ili9xxx_data *data = dev->data;
+	struct display_buffer_descriptor mipi_desc;
+	int r;
+	uint32_t gram_data, read_cnt, nbr_of_reads;
+	uint16_t *read_data_start = (uint16_t *)buf;
+
+	if (data->pixel_format != PIXEL_FORMAT_RGB_565) {
+		/* Only RGB565 can be supported, see note below */
+		return -ENOTSUP;
+	}
+
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller than width");
+	__ASSERT((desc->pitch * data->bytes_per_pixel * desc->height) <=
+			 desc->buf_size,
+		 "Output buffer to small");
+
+	LOG_DBG("Reading %dx%d (w,h) @ %dx%d (x,y)", desc->width, desc->height,
+		x, y);
+
+	r = ili9xxx_set_mem_area(dev, x, y, desc->width, desc->height);
+	if (r < 0) {
+		return r;
+	}
+
+	/*
+	 * ILI9XXX stores all pixel data in graphics ram (GRAM) as 18 bit
+	 * values. When using RGB565 pixel format, pixels are converted to
+	 * 18 bit values via a lookup table. When using RGB888 format, the
+	 * lower 2 bits of each pixel are simply dropped. When reading pixels,
+	 * the response format will always look like so:
+	 * | R[5:0] | x | x | G[5:0] | x | x | B[5:0] | x | x |
+	 * Where x represents "don't care". The internal format of the
+	 * ILI9XXX graphics RAM results in the following restrictions:
+	 * - RGB888 mode can't be supported.
+	 * - we can only read one pixel at once (since we need to do
+	 *   byte manipulation on the output)
+	 */
+
+	/* Setup MIPI descriptor to read 3 bytes (one pixel in GRAM) */
+	mipi_desc.width = 1;
+	mipi_desc.height = 1;
+	/* Per MIPI API, pitch must always match width */
+	mipi_desc.pitch = 1;
+
+	nbr_of_reads = desc->width * desc->height;
+
+	/* Initial read command should consist of RAMRD command, plus
+	 * 8 dummy clock cycles
+	 */
+	uint8_t cmd[] = {ILI9XXX_RAMRD, 0xFF};
+
+	for (read_cnt = 0; read_cnt < nbr_of_reads; read_cnt++) {
```

Comment:
Why not `for (uint32_t read_cnt = 0; read_cnt < nbr_of_reads; read_cnt++) {`?
read_cnt is not used anywhere else.

Reply:
no reason, I was just grouping declarations at top of the function. I'll update this